js/iframe-coordinator/libs/client.ts

```diff
@@ -23,10 +23,45 @@ function start(expectedOrigin) {
         msg: location
       });
     });
+
+    /**
+     * Request a toast message be displayed by the host.
+     * 
+     * The page embedding the host is responsible for handling the fired custom event and
+     * presenting/styling the toast.  Application-specific concerns such as level, TTLs,
+     * ids for action callbacks (toast click, toast action buttons), etc. can be passed via custom
+     * properties of the options param.
+     * 
+     * @param {string} message - The message content of the toast
+     * @param {object=} options - Supplimental toast options.
+     * @param {string=} options.title - Optional title for the toast.
+     * @param {*=} options.x - Optional, application-specific toast properties.  Note: Must be JSON serializable.
+     * 
+     * @example
+     * worker.requestToast('Hello world');
+     * 
+     * @example
+     * worker.requestToast('World', {title: 'Hello'});
+     * 
+     * @example
+     * worker.requestToast('World', {title: 'Hello', ttl: 5, level: 'info'});
+     */
+    worker.requestToast = function (message: string, {title = null, ...custom}: undefined | ToastOptions = {}): void {
```

Comment:
Originally, the signature was `(message, title=null, custom={})`.

The more I thought about it though, the more I liked a bag of options.  Makes adding custom props easier, especially if the toast is untitled.  Open for opinions here.

The other minor issue with this is that you pass the title in as an object, but it's fired out as a first class property of the event.  `event.detail.[title|message|custom]`.  I can smooth this out either direction, but wanted to open it up to the team.

Reply:
I should probably take a look at our typescript/babel settings just to make sure we're transpiling everything I'm expecting.

Reply:
I think a signature of `(message, {title = null, custom = null})` would be better. I'm fine putting custom data in the same structure as other optional parameters, but let's give it it's own dedicated space in the object so we can evolve the API without breaking it.

I'm actually wondering if it might not be better to just expose something that doesn't translate the messages into Elm so much, but I think I will defer that question until we start to think about versioning the API.